### PR TITLE
chain: clean up oracle module code

### PR DESCRIPTION
**Pull request by sorawit**
Also make sure that OnRecvPacket return errors on unknown packs.

### CODE REVIEW

**Review by Benzbeeb — APPROVED** [review]

File: chain/x/oracle/module.go [hunk 1]
```diff
@@ -49,199 +56,149 @@ func (AppModuleBasic) ValidateGenesis(cdc codec.JSONMarshaler, bz json.RawMessag
...
-			requestData.MinCount, requestData.ClientID,
-			sdk.AccAddress([]byte("Unknown")),
-		)
-		return handleMsgRequestDataIBC(
-			ctx, am.keeper, newMsg, packet.GetDestPort(), packet.GetDestChannel(),
-		)
+// OnRecvPacket implements ics-26 IBCModule interface.
+func (am AppModule) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet) (*sdk.Result, error) {
+	var req OracleRequestPacketData
+	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &req); err != nil {
```
Comment (sorawit) [inline chain/x/oracle/module.go:177]: Note this is the only non-trivial change. I make it return error if message parsing fails.

File: chain/x/oracle/module.go [hunk 2]
```diff
@@ -12,33 +12,40 @@ import (
...
 	ibctypes "github.com/cosmos/cosmos-sdk/x/ibc/types"
 
 	"github.com/cosmos/cosmos-sdk/client/context"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	abci "github.com/tendermint/tendermint/abci/types"
 )
 
-// AppModule Basics object
+var (
+	_ module.AppModule      = AppModule{}
```
Comment (taobun) [inline chain/x/oracle/module.go:29]: Cosmos-sdk write these line. But why we need these?

File: chain/x/oracle/module.go [hunk 3]
```diff
@@ -49,199 +56,149 @@ func (AppModuleBasic) ValidateGenesis(cdc codec.JSONMarshaler, bz json.RawMessag
...
+	counterparty channeltypes.Counterparty, version string,
 ) error {
-	// // TODO: Enforce ordering, currently relayers use ORDERED channels
-	// if version != types.Version {
-	// 	return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "invalid version: %s, expected %s", version, "ics20-1")
-	// }
-
-	// Claim channel capability passed back by IBC module
-	if err := am.keeper.ScopedKeeper.ClaimCapability(ctx, chanCap, ibctypes.ChannelCapabilityPath(portID, channelID)); err != nil {
+	// Claim channel capability passed back by IBC module.
```
Comment (taobun) [inline chain/x/oracle/module.go:131]: Remove version?

File: chain/x/oracle/module.go [hunk 4]
```diff
@@ -49,199 +56,149 @@ func (AppModuleBasic) ValidateGenesis(cdc codec.JSONMarshaler, bz json.RawMessag
...
-		newMsg := NewMsgRequestData(
-			requestData.OracleScriptID, calldata, requestData.AskCount,
-			requestData.MinCount, requestData.ClientID,
-			sdk.AccAddress([]byte("Unknown")),
-		)
-		return handleMsgRequestDataIBC(
-			ctx, am.keeper, newMsg, packet.GetDestPort(), packet.GetDestChannel(),
-		)
+// OnRecvPacket implements ics-26 IBCModule interface.
+func (am AppModule) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet) (*sdk.Result, error) {
```
Comment (taobun) [inline chain/x/oracle/module.go:175]: We need to cal packet executed?

**Review by taobun — CHANGES_REQUESTED** [review]

File: chain/x/oracle/module.go (hunk 2 above)
Comment (sorawit) [reply chain/x/oracle/module.go:29]: It's not strictly necessary, but it makes compile error messages easier to follow.

File: chain/x/oracle/module.go (hunk 3 above)
Comment (sorawit) [reply chain/x/oracle/module.go:131]: Yeah I don't think it's needed for us. Let's just skip it.

File: chain/x/oracle/module.go (hunk 4 above)
Comment (sorawit) [reply chain/x/oracle/module.go:175]: Don't think so. That will require we also define ack packet. Let's just say for now we don't follow ICS ack standard.

**Review by taobun — APPROVED** [review]

**Pull request merged**